### ISSUE TITLE
bot not merging although all conditions are ok

### ISSUE THREAD

**Issue by KITSAndrzejWower** [bobvanderlinden/probot-auto-merge#145]
Hi 
just started testing the bot and it seems to not work with basic scenario https://github.com/KITSAndrzejWower/bot-test/

also where is reporting ?

I have just created 3 PRs hoping that they will merg one after another and update them selves. 
Is there a way to see why queue is not moving ?

Thank you

**Comment by KITSAndrzejWower** [comment]
Seems like its stuck after updating the branch ...

**Comment by bobvanderlinden** [comment]
Yes, I see errors coming in from your repository:

```
HttpError: {"message":"Bad credentials","documentation_url":"https://developer.github.com/v4"}
```

I still do not know why these are happening in such large numbers for _some_ repositories and not for others. If it is a bug in probot-auto-merge, I'd guess it's in the queueing system and installation_ids of received events get mixed up. I need to give this another good look to make sure the problem isn't residing here.

It's hard for me to reproduce, since the PRs in my private repo and the ones in probot-auto-merge seem to be working fine. Even when queueing multiple after eachother. See for instance the last 3 package updates: https://github.com/bobvanderlinden/probot-auto-merge/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed

Another thing that I'm suspecting of not being stable is the free tier of Heroku. I can imagine that some problems are due to the time auto-merge needs to stay up in order to handle all branch updates and merges.

It might help to host your own instance of probot-auto-merge and see whether that helps.

**Comment by KITSAndrzejWower** [comment]
Ah ok Thanks for the response anyway we needed a bit different bot. It needs to rebase in 'git' way then run the tests and if pass merge. Its safer. If you are interested something like https://github.com/depop/github-merge-bot

**Comment by bobvanderlinden** [comment]
For that you need to enable `Require branches to be up to date before merging` in Github:

<img width="989" alt="branch protection rule 2019-02-05 10-47-40" src="https://user-images.githubusercontent.com/6375609/52265111-9fa56b80-2933-11e9-9953-cfb383bd760d.png">

auto-merge is compatible with this option and also with the check requirements. I use this in my repos as well. This way it doesn't matter if you want to use rebases or merges, both will work as both can be checked whether the branch is up-to-date.

**Comment by tracom1** [comment]
Can you maybe explain a little bit more about how auto-merge is compatible with requiring status checks?  We're having an issue with a forked version of your repo where the auto-merges aren't working, and the only difference I can tell from one repo to another (where it does work) are these branch protection conditions.

**Comment by tracom1** [comment]
The only error we're seeing is that "the pull request is blocked by a failing or missing check".  But if we wait some time until the build finishes successfully, and then just remove and re-add the same label, it will then merge automatically.

**Comment by fahmisdk6** [comment]
Could this be a problem with webhook?

**Comment by bobvanderlinden** [comment]
This could've been already fixed by the changes regarding status events. @tracom1 @KITSAndrzejWower could you confirm this issue is still relevant?

**Comment by tracom1** [comment]
This ticket is no longer relevant for me, and was solved in https://github.com/bobvanderlinden/probot-auto-merge/issues/266

**Comment by bobvanderlinden** [comment]
Thanks for the confirmation. Closing this one.
